From ff3b3de16df812ca1e5d41a3d3b50d7a8fd1eb75 Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Sun, 25 Feb 2024 10:23:29 -0500
Subject: [PATCH 14/14] When reading SMSC, ignore invalid SMSC

On Samsung, when reading SMSC we get weird strings like
"4392084000100",149
The first number is nowhere close to the SMSC phone number
This number is used when sending SMS, and since it is invalid, sending
SMS doesn't work.
We have no idea what this value means or anything, but let's just ignore
it. In 2024 we can send SMS just fine without a SMSC

This commit fixes sending SMS on OrangeF on my S24
---
 src/java/com/android/internal/telephony/MessagingResponse.java | 2 ++
 src/java/com/android/internal/telephony/RadioResponse.java     | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/src/java/com/android/internal/telephony/MessagingResponse.java b/src/java/com/android/internal/telephony/MessagingResponse.java
index 19211e15f5..3ddf6762dd 100644
--- a/src/java/com/android/internal/telephony/MessagingResponse.java
+++ b/src/java/com/android/internal/telephony/MessagingResponse.java
@@ -170,6 +170,8 @@ public class MessagingResponse extends IRadioMessagingResponse.Stub {
      * @param smsc Short Message Service Center address on the device
      */
     public void getSmscAddressResponse(RadioResponseInfo responseInfo, String smsc) {
+        if(smsc.contains("\"") || smsc.contains(","))
+            smsc = "";
         RadioResponse.responseString(HAL_SERVICE_MESSAGING, mRil, responseInfo, smsc);
     }
 
diff --git a/src/java/com/android/internal/telephony/RadioResponse.java b/src/java/com/android/internal/telephony/RadioResponse.java
index 71bb2e5b75..87cb4ad615 100644
--- a/src/java/com/android/internal/telephony/RadioResponse.java
+++ b/src/java/com/android/internal/telephony/RadioResponse.java
@@ -1267,6 +1267,8 @@ public class RadioResponse extends IRadioResponse.Stub {
      * @param smsc Short Message Service Center address on the device
      */
     public void getSmscAddressResponse(RadioResponseInfo responseInfo, String smsc) {
+        if(smsc.contains("\"") || smsc.contains(","))
+            smsc = "";
         responseString(responseInfo, smsc);
     }
 
-- 
2.34.1

